internal/core/src/exec/operator/groupby/SearchGroupByOperator.h

```diff
@@ -97,30 +190,71 @@ class SealedDataGetter : public DataGetter<T> {
                     auto pw = segment_.chunk_view<std::string_view>(field_id_,
                                                                     chunk_id);
-                    pw_map_[chunk_id] = std::move(pw);
+                    str_pw_map[chunk_id] = std::move(pw);
                 }
-                auto& pw = pw_map_[chunk_id];
+                auto& pw = str_pw_map[chunk_id];
                 auto& [str_chunk_view, valid_data] = pw.get();
                 if (!valid_data.empty() && !valid_data[inner_offset]) {
                     return std::nullopt;
                 }
                 std::string_view str_val_view = str_chunk_view[inner_offset];
                 return std::string(str_val_view.data(), str_val_view.length());
+            } else if constexpr (std::is_same_v<InnerRawType, milvus::Json>) {
+                if (json_pw_map.find(chunk_id) == json_pw_map.end()) {
+                    auto pw =
+                        segment_.chunk_view<milvus::Json>(field_id_, chunk_id);
+                    json_pw_map[chunk_id] = std::move(pw);
+                }
+                auto& pw = json_pw_map[chunk_id];
+                auto& [json_chunk_view, valid_data] = pw.get();
+                if (!valid_data.empty() && !valid_data[inner_offset]) {
+                    return std::nullopt;
+                }
+                auto& json_val = json_chunk_view[inner_offset];
+                // Handle different output types using macro for cleaner code
+                JSON_TYPE_CASE(
+                    OutputType, bool, bool, static_cast<bool>(result.value()))
+                JSON_TYPE_CASE(OutputType,
+                               int8_t,
+                               int64_t,
+                               static_cast<int8_t>(result.value()))
+                JSON_TYPE_CASE(OutputType,
+                               int16_t,
+                               int64_t,
+                               static_cast<int16_t>(result.value()))
+                JSON_TYPE_CASE(OutputType,
+                               int32_t,
+                               int64_t,
+                               static_cast<int32_t>(result.value()))
+                JSON_TYPE_CASE(OutputType,
+                               int64_t,
+                               int64_t,
+                               static_cast<int64_t>(result.value()))
+                // Special handling for string type
+                JSON_STRING_CASE(OutputType)
+                return std::nullopt;
             } else {
-                auto pw = segment_.chunk_data<T>(field_id_, chunk_id);
+                static_assert(
+                    std::is_same_v<OutputType, InnerRawType>,
+                    "OutputType and InnerRawType must be the same for "
+                    "non-json/string field group by");
+                auto pw =
+                    segment_.chunk_data<InnerRawType>(field_id_, chunk_id);
                 auto& span = pw.get();
```

Comment:
is json_cast_type is given but ran across wrong-type data, we will throw exception directly